sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -234,8 +235,11 @@ internal class DefaultExperimentClient internal constructor(
                     Logger.d("Received fetch response: $response")
                     val variants = parseResponse(response)
                     future.complete(variants)
-                } catch (e: IOException) {
-                    onFailure(call, e)
+                } catch (e: Exception) {
+                    when (e) {
+                        is IOException -> onFailure(call, e)
+                        is JSONException -> future.completeExceptionally(e)
+                    }
```

Comment:
onFailure is the same as calling completeExceptionally, so we should just maintain consistency. Further, this logic would ignore all other errors, which is not ideal. Either we (a) catch and propagate all exceptions or (b) only catch and propagate some exceptions, leavin some to be thrown in the callback. 

I think we go with (a) so we can just do catch any exception and pass it to onFailure

Reply:
On this, the onFailure function in the Callback interface has signature `fun onFailure(call: Call, e: IOException)`, should I create an onFailure function which takes in any exception type then?